docs/adr/0005-api-for-spaces.md

```diff
@@ -0,0 +1,129 @@
+
+### Get "Home folder"
+
+Retrieve information about the home space of a user. Note: The user has access to more spaces. This call only returns the home space to provide API parity with the Graph API.
+
+API Call: `/me/drive`: Returns the information about the users home folder.
+
+### Get All Spaces of a User
+
+Retrieve a list of available spaces of a user. This includes all spaces the user has access to at that moment, also the home space.
+
+API Call: `/me/drives`: Returns a list of spaces.
+
+There is also `/drives`, returning the list of spaces the user has access to. This endpoint is used to access any space by id using `/drives/{drive-id}`.
+
+### Common Reply
+
+The reply to both calls is either one or a list of [Drive representation objects](https://docs.microsoft.com/de-de/graph/api/resources/drive?view=graph-rest-1.0):
+
+```
+{
+  "id": "string",
+  "createdDateTime": "string (timestamp)",
+  "description": "string",
+  "driveType": "personal | projectSpaces | shares",
+  "oCDriveStatus": "accepted | pending | mandatory | offline",
+  "lastModifiedDateTime": "string (timestamp)",
+  "name": "string",
+  "owner": { "@odata.type": "microsoft.graph.identitySet" },
+  "quota": { "@odata.type": "microsoft.graph.quota" },
+  "root":  { "@odata.type": "microsoft.graph.driveItem" },
+  "webUrl": "url",
+  "oCCoOwner": [ { "@odata.type": "microsoft.graph.identitySet" } ]
+}
+
+```
+
+The meaning of the object are in ownClouds context:
+
+1. **id** - a unique ID identifying the space
+2. **driveType** - describing the type of the space.
+3. **oCDriveStatus** - telling the status (*)
+4. **owner** - an owner object to whom the space belongs
+5. **quota** - quota information about this space
+6. **root**  - the root driveItem object.
+7. **webUrl** - The URL to make this space visible in the browser.
+8. **oCCoOwner** - optional array owner objects of the co-owners of a space (*)
+
+> Note: the **root** object is a [driveItem](https://docs.microsoft.com/de-de/graph/api/resources/driveitem?view=graph-rest-1.0) and contains information about the root of the space. Especially important is the member `webDavUrl` in the root object which provides the full URL to be used by clients to address items within the space. Also it contains the fields `eTag` and `cTag`.
+
+The following *driveType* values are available in the first step, but might be enhanced later:
+
+* **personal**: The users home space
+* **projectSpaces**: The project spaces available for the user (*)
+* **shares**: The share jail, contains all shares for the user (*)
+
+Other space types such as backup, hidden etc. can be added later as requested.
+
+The (*) marked types are not defined in the official MS API. They are prefixed with `oC` to avoid namespace clashes.
+
+
+The following *driveStatus* values are available:
+
+* **accepted**: The user has accepted the space and uses it
```

Comment:
These fields represents only a perspective on how shares could work.
As a user when I'm receiving a share I may want to check the contents of it before accepting it. Accepts has collateral effect that the share will be  synchronised in the clients. 
As a user you may want to access the shares but not necessarily sync it.